tests/Feature/AdvisorGenerationTest.php

```
@@ -51,161 +58,154 @@ public function test_deterministic_template_loading_and_variable_substitution()
+        // Mock all LLM calls with responses that include "Think step by step"
+        $callCount = 0;
         $this->mockLLMService
             ->shouldReceive('generateText')
-            ->withArgs(function($prompt, $options) {
-                return str_contains($prompt, 'Project Knowledge') &&
-                       $options['model'] === 'o3-deep-research';
-            })
-            ->andReturn('Generated PK content');
-            
+            ->atLeast(1)
+            ->andReturnUsing(function ($prompt, $options) use ($enhancedTemplate, &$callCount) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$options'. (Unused Code Rules)

(UnusedFormalParameter)